JMH annotation

bgprudhomme opened the pull request.

bgprudhomme committed:
Begin JMH annotation

bgprudhomme committed:
Execute jar

khatchad commented on .travis.yml:
```diff
@@ -19,3 +19,5 @@ env:
 script:
   - mvn clean org.jacoco:jacoco-maven-plugin:prepare-agent install
+  - mvn clean package
+  - java -jar target/java-webserver-1.0.0.jar
```
No changes in the CI for now, please. I'll let you know.

khatchad commented on pom.xml:
```diff
@@ -141,6 +141,17 @@
+        <dependency>
+	    <groupId>org.openjdk.jmh</groupId>
+	    <artifactId>jmh-core</artifactId>
+	    <version>1.32</version>
```
Good. Is this the latest version?

bgprudhomme replied:
No - looks like it's actually 1.37: https://mavenlibs.com/maven/dependency/org.openjdk.jmh#:~:text=jmh%20%2D%20All%20Dependencies-,jmh%2Dcore%201.37,-%40org.openjdk.jmh
I'll update it now

khatchad commented on pom.xml:
```diff
@@ -141,6 +141,17 @@
+	    <version>1.32</version>
+	</dependency>
+	<dependency>
+	    <groupId>org.openjdk.jmh</groupId>
```
Do we need both of these?

khatchad commented on src/test/java/au/id/deejay/webserver/server/WebServerExecutorTest.java:
```diff
@@ -12,24 +13,25 @@
 	}
 
-	@Test
+	@Benchmark
```
Good.

khatchad commented on src/test/java/au/id/deejay/webserver/server/WebServerExecutorTest.java:
```diff
@@ -12,24 +13,25 @@
 	}
 
-	@Test
+	@Benchmark
```
You're gonna need arguments here. What do the arguments to the `WebServerExecutor()` ctor below mean?

bgprudhomme replied:
https://github.com/djessup/java-webserver/blob/975ed887b8044edb81cd9e5a294c2ff590ec6256/src/main/java/au/id/deejay/webserver/server/WebServerExecutor.java#L40-L47

khatchad replied:
Oh, so I don't think we'll need any of these ,right? We won't need max threads anymore, correct?

bgprudhomme replied:
Right, because newVirtualThreadPerTaskExecutor() doesn't take a thread amount as a parameter

khatchad replied:
So, for this test, how would we increase the number of transactions being processed?

khatchad commented on src/test/java/au/id/deejay/webserver/server/WebServerTest.java:
```diff
@@ -19,7 +20,8 @@
  * @author David Jessup
  */
-@RunWith(MockitoJUnitRunner.class)
+//@RunWith(MockitoJUnitRunner.class)
```
I would think you're still going to need Mockito. These tests may not run, otherwise. Am I wrong?

Also, please delete instead of comment.

khatchad replied:
On second thought, maybe we should treat Mockito like JUnit and also replace it with JMH.

khatchad reviewed (changes requested):
Please remove lines instead of commenting them out.

bgprudhomme committed:
Update .travis.yml

bgprudhomme committed:
Update pom.xml

bgprudhomme committed:
Update WebServerTest.java

bgprudhomme committed:
Update pom.xml

bgprudhomme committed:
Update pom.xml

bgprudhomme committed:
Add MockitoAnnotations

khatchad commented on pom.xml:
```diff
@@ -141,6 +141,17 @@
+        <dependency>
+	    <groupId>org.openjdk.jmh</groupId>
+	    <artifactId>jmh-core</artifactId>
+	    <version>1.37</version>
```
Add constant like in https://github.com/ponder-lab/Stream-Performance-Evaluation/blob/4c67246f6789e392d8a64b2d7d3da1d0555af035/performance/java-design-patterns/java-design-patterns.patch#L1151

khatchad commented on pom.xml:
```diff
@@ -141,6 +141,17 @@
+	<dependency>
+	    <groupId>org.openjdk.jmh</groupId>
+	    <artifactId>jmh-generator-annprocess</artifactId>
+	    <version>1.37</version>
```
Use the constant.

khatchad commented on pom.xml:
```diff
@@ -141,6 +141,17 @@
+	    <groupId>org.openjdk.jmh</groupId>
+	    <artifactId>jmh-generator-annprocess</artifactId>
+	    <version>1.37</version>
+	</dependency>
```
Do we also need a scope here? Like this? https://github.com/ponder-lab/Stream-Performance-Evaluation/blob/4c67246f6789e392d8a64b2d7d3da1d0555af035/performance/java-design-patterns/java-design-patterns.patch#L1169

khatchad committed:
Update pom.xml

Remove whitespace.

khatchad committed:
A bit of formatting.

khatchad committed:
Remove whitespace.

khatchad committed:
Downgrade the JMH version.

khatchad committed:
Additional JMH arguments.

From: https://github.com/guozheng/jmh-tutorial/blob/master/README.md#jmh-command-line-options

khatchad committed:
Remove benchmarks from tests that test exceptions.

khatchad committed:
Ignore JMH tests.

khatchad committed:
Revert "Downgrade the JMH version."

This reverts commit d6781ad63c535b5875b6079ef62f7ec00b4befed.

khatchad committed:
Skip tests for now.

khatchad committed:
Open and close mocks during JMH tests.
